See http://cygwin.com/ml/cygwin-patches/2008-q4/msg00022.html

--- origsrc/gnome-vfs-2.24.0/configure.in	2008-12-08 11:16:08.862375000 -0600
+++ src/gnome-vfs-2.24.0/configure.in	2008-12-08 16:45:20.128000000 -0600
@@ -817,6 +817,7 @@
 	LIBS="$LIBS -lresolv"
 	AC_MSG_CHECKING(for res_search in -lresolv)
 	AC_LINK_IFELSE([[
+		#include <netinet/in.h>
 		#include <resolv.h>
 		int main()
 		{
@@ -841,6 +842,7 @@
         	  #ifdef linux
         	  #define _BSD_SOURCE 1
         	  #endif
+        	  #include <netinet/in.h>
         	  #include <resolv.h>
         ], [
 		int foo = res_ninit (&_res);
@@ -856,8 +858,8 @@
 else
     AC_CHECK_LIB(bind, res_ninit, 
     	[AC_DEFINE(HAVE_RES_NINIT, 1, [Indicates if res_ninit is available])
-	 RESOLVER_LIBS="-lbind",
-    	 AC_CHECK_LIB(resolv, res_ninit, 
+	 RESOLVER_LIBS="-lbind"],
+    	[AC_CHECK_LIB(resolv, res_ninit, 
 		[AC_DEFINE(HAVE_RES_NINIT, 1, [Indicates if res_ninit is available])
 		 if test "x$need_lib_resolv" = "xno"; then
 			RESOLVER_LIBS="-lresolv"
@@ -1017,13 +1019,15 @@
 					    LIBS="$LIBS -l$lib"
 					    have_gethostbyname=yes
 					    have_resolver=yes
-
-					    AC_DEFINE(USE_GETHOSTBYNAME, 1,
-							 [Define if gethostbyname is used for name resolution])
 					    break
 		])
       done
    fi
+
+   if test $have_gethostbyname = yes; then
+     AC_DEFINE(USE_GETHOSTBYNAME, 1,
+               [Define if gethostbyname is used for name resolution])
+   fi
 fi
     
 dnl ==============================================================================
